.github/workflows/main.yml

```diff
@@ -28,6 +28,32 @@ jobs:
         git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
         git config --global user.name "github-actions[bot]"
 
+    - name: Install Linux system dependencies
+      if: runner.os == 'Linux'
+      run: |
+        sudo apt-get install -y \
+          clang \
+          rpm \
+          doxygen \
+          xdot \
+          graphviz
+
+        # Use Homebrew for latest build tool versions
+        brew install \
```

Comment:
What's the advantage of using Homebrew on Linux like this?

Reply:
Ah, is it because Homebrew offers newer versions? If so, why are we interested in getting new versions of CMake, CCache and Ninja but not the other dependencies?

Reply:
> is it because Homebrew offers newer versions?

Yes, for CI, it helps to use the latest versions of some tools for their enhancements/updates to speed and stability. It's also fine because those Homebrew-installed tools aren't _required_ (as listed in the README) to actually build the project.

> why are we interested in getting new versions of CMake, CCache and Ninja but not the other dependencies?

Actually, ideally we would pin the CMake version to our `cmake_minimum_required` version, so I will be making that change for Linux only as a way to test minimum version, and then macOS uses latest version to test that as well.

I didn't use Homebrew to install the other tools listed under `apt install` because we want to ensure that those tool versions included by default on this version of Ubuntu (20.04) are sufficient for the building and testing of sleigh---it's like a lower-bound version support because the macOS CI runner uses Homebrew for everything and thus represents a latest version support, for the most part.

Reply:
Sounds great! Thanks for elaborating.